dns.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net"
+)
+
+const consulDnsAddr = "127.0.0.1:8600"
+const datacenter = "dc1"
+
+func NewConsulDNS() *ConsulDNS {
+	return &ConsulDNS{
+		r: &net.Resolver{
+			PreferGo: true,
+			Dial: func(ctx context.Context, network, address string) (net.Conn, error) {
+				d := net.Dialer{}
+				return d.DialContext(ctx, "udp", consulDnsAddr)
+			},
+		},
+		ctx: context.Background(),
+	}
+}
+
+type ConsulDNS struct {
+	r   *net.Resolver
+	ctx context.Context
+}
+
+func (c *ConsulDNS) GetServiceAddr(serviceName string) (addr string, err error) {
+	name := fmt.Sprintf("%s.service.%s.consul", serviceName, datacenter)
```

Comment:
String concatenation is actually faster than format printing. [example](https://dev.to/pmalhaire/concatenate-strings-in-golang-a-quick-benchmark-4ahh)